testcases/kernel/syscalls/adjtimex/adjtimex02.c

```diff
@@ -10,115 +10,188 @@
 #include <unistd.h>
 #include <pwd.h>
 #include "tst_test.h"
+#include "lapi/syscalls.h"
 
 #define SET_MODE ( ADJ_OFFSET | ADJ_FREQUENCY | ADJ_MAXERROR | ADJ_ESTERROR | \
-	ADJ_STATUS | ADJ_TIMECONST | ADJ_TICK )
+				ADJ_STATUS | ADJ_TIMECONST | ADJ_TICK )
 
-static int hz;			/* HZ from sysconf */
-
-static struct timex *tim_save;
-static struct timex *buf;
+static int hz;		/* HZ from sysconf */
 
+static struct timex *tim_save, *buf;
 static struct passwd *ltpuser;
 
-static void verify_adjtimex(unsigned int nr)
+static struct test_case{
+	unsigned int modes;
+	long lowlimit;
+	long highlimit;
+	long delta;
+	int exp_err;
+	int droproot;
+};
+
+enum tst_timex_type
+{
+	TST_KERN_OLD_TIMEX,
+	TST_KERN_TIMEX
+};
+
+static inline int sys_adjtimex (void *timex)
+{
+	return tst_syscall (__NR_adjtimex, timex);
+}
+
+static struct test_case tc[] = {
+	{
+	.modes= SET_MODE,
+	.exp_err = EFAULT,
+	},
+	{
+	.modes = ADJ_TICK,
+	.lowlimit = 900000,
+	.delta = 1,
+	.exp_err = EINVAL,
+	},
+	{
+	.modes = ADJ_TICK,
+	.highlimit = 1100000,
+	.delta = 1,
+	.exp_err = EINVAL,
+	},
+	{
+	.modes= SET_MODE,
+	.exp_err = EPERM,
+	.droproot = 1,
+	},
+	{
+	.modes = ADJ_OFFSET,
+	.highlimit = 512000L,
+	.delta = 1,
+	.exp_err = EINVAL,
+	},
+	{
+	.modes = ADJ_OFFSET,
+	.lowlimit = -512000L,
+	.delta = -1,
+	.exp_err = EINVAL,
+	},
+};
+
+static struct test_variants
+{
+	int (*adjtimex) (void *timex);
+	enum tst_timex_type type;
```

Comment:
This type is not used at all, no need to add it.

Reply:
Removed.